### ISSUE TITLE
Auto layers aliases are not resolved relatively in SCSS files and server files.

### ISSUE THREAD

**Issue by Dananz** [nuxt/nuxt#24427]
### Environment

- Operating System: `Darwin`
- Node Version:     `v18.18.2`
- Nuxt Version:     `3.8.2`
- CLI Version:      `3.10.0`
- Nitro Version:    `2.8.0`
- Package Manager:  `pnpm@8.7.1`
- Builder:          `-`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

1. start the app (wait until installation is done and run `$ npm run start:dev`)
2. should throw an error for `@import '@/styles/global';`

### Describe the bug

Whats really happening is:
when a layer imports stuff using `@/` it wont auto use aliases to resolve the files relatively to the layer root. It will look the file under the **extending** layer instead.

### Additional context

For this current reproduction, SCSS files are not imported correctly,  I couldn't reproduce it, but for my local project Nuxt throws error for server folder imports too.

I added an example of how I use the server in my local app. its under `apps/site-two/server/utils/utils.ts`, when trying to import `import { anotherUtilFunction } from '@/lib/helpers';` it throws an error that its not exists under the **extending** layer

### Logs

```shell-script
[2:30:30 PM]  ERROR  [sass] ENOENT: no such file or directory, open '/home/projects/nuxt-layers-wrong-imports/apps/site-two/src/styles/global'
  ╷
2 │ @import '@/styles/global';
  │         ^^^^^^^^^^^^^^^^^
  ╵
  ../../packages/ui/components/UiButton.vue 2:9  root stylesheet

# as you can see, it looks for the styles folder inside `site-two` when it should look for it under `packages/ui`
```

**Comment by danielroe** [comment]
related: https://github.com/nuxt/nuxt/pull/19986.

We likely need to provide a similar plugin to nitro (or implement upstream if appropriate) - and also maybe a postcss resolver....

**Comment by janniksohn** [comment]
@danielroe 
Any updates or ideas to avoid this problem?

I had the idea to inject the absolute path as an scss variable and use that before every import, but that doesnt seem to work

### nuxt.config.ts
```typescript
import { fileURLToPath } from 'url'
import { dirname, join } from 'path'

const currentDir = dirname(fileURLToPath(import.meta.url))

export default defineNuxtConfig({
  vite: {
    css: {
      preprocessorOptions: {
        scss: {
          additionalData: `$base-path: "${currentDir}";`,
        }
      }
    }
  },
})
```

### SomeComponent.vue
```vue
<script setup lang="ts">
defineProps<{
  link: string,
}>();
</script>

<template>
  <NuxtLink v-if="link" :to="link">
    <slot />
  </NuxtLink>
</template>

<style lang="scss" scoped>
@import url($base-path + '/assets/styles/main/1_base/_variables.scss');

a:deep(.image) {
  overflow: hidden;

  img {
    // use variable from scss partial here
    transition: scale $transition;
  }
}
 </style>
```

**Comment by danielroe** [comment]
For a solution, I believe the suggestion I've provided:

> We likely need to provide a similar plugin to nitro (or implement upstream if appropriate) - and also maybe a postcss resolver....

**Comment by janniksohn** [comment]
for my specific issue I already found a solution, I was just dumb. Instead of `@import` in the Component itself i added it as the vite `preprocessorOption` and then its correctly resolved

**Comment by Kapcash** [comment]
I have a use case which I think is related.
So here is my update on this issue, as of today, using Nuxt v3.17.5.

TLDR;
* `scss` files override by layers doesn't work in the `components/` folder, but it does work in the `assets` folder.
* `#layers/` alias import doesn't work in `scss` files.
* Importing scss file from the Vue SFC style block targets the correct overridden file only with the `assets/` default alias. `@/assets` always target the layer files.

-----

Setup:

```
`- apps/baselayer/
  `- components/
    `- Counter.vue // imports Counter.scss
    `- Counter.scss
`- apps/clientapp
  `- components/
    `- Counter.scss
```

Goal: override the scss file only, not the full component, from the `clientapp`.

What I've tried so far:
- ❌ Using `assets` alias: `<style scoped src="assets/scss/Counter.scss" />`.
- ❌ Using @ alias path import: `<style scoped src="@/assets/scss/Counter.scss" />`.

Both only load the layer's `scss` file. Like if the path was relative from the layer's component file. `apps/clientapp/components/Counter.scss` is completely ignored.

Result: **The layer file override for `scss` files doesn't work in the `components/` folder.**

--------

When moving the `scss` files to the `assets` folder:

```
`- apps/baselayer/
  `- components/
    `- Counter.vue // imports Counter.scss
  `- assets/scss/
    `- Counter.scss
`- apps/clientapp
  `- assets/scss/
    `- Counter.scss
```

- ❌ Using `@` alias path import: `<style scoped src="@/assets/scss/Counter.scss" />`. It imports only the one from the layer.
- ✅ Using `assets/` alias path import: `<style scoped src="assets/scss/Counter.scss" />` It imports the one from the correct project (either baselayer or the clientapp).

Result: **The layer file override for `scss` files work in the `assets/` folder.**

Note: I haven't tried with a custom folder like `/styles` to see if it acts different.

------

Using this last technic to override the .scss file, I've tried to import the baselayer scss file from the overriding file in clientapp:

```scss
// apps/clientapp/assets/scss/Counter.scss
@use '#layers/baselayer/assets/scss/Counter';
```
But it gives this error:
```
[sass] The default namespace "" is not a valid Sass identifier.
```

Adding a namespace like `@use '#layers/baselayer/assets/scss/Counter' as count;`, it gives:
```
[sass] Module loop: this module is already being loaded.
```

Like if the module was trying to import itself.

Using a relative path import from the clientapp/assets/scss/Counter.scss does work though:

```scss
// apps/clientapp/assets/scss/Counter.scss
@use '../../../baselayer/assets/scss/Counter';
```

The behaviors I expected are:
* any file in the `components/` folder would be overridden by the layer when the files have the same path/name, whether they are `.vue` or any other files.
* `#layers/` alias work in `scss` files
* `@/` alias should target either layer or client app when used in the `<style src="">` block in vue SFCs.

I hope this information is useful. Let me know if you need any reproduction repository :)
